### DISCUSSION TITLE
Graph view in databaseviewer

### DISCUSSION THREAD

**Discussion by SiinaIng** [introlab/rtabmap#976] [Q&A]
Hi @matlabbe!

When I change any core parameter in databaseviewer, it seems to do nothing to the graph. How do I update the changes to the graph view?

**Comment by matlabbe** [comment]
If it is a graph optimization related parameters, you can just click/unclick the checkbox next to the horizontal bar in Graph View (or the oher one labeled "Span to all maps"). This should force re-optimizing with current parameters.

**Reply by SiinaIng** [reply]
So only changing the optimizer parameters will yield any change to the graph? (I'm very new to this topic and using RTAB-map) Will changing ICP parameters have any result in the graph or the 3D model?

**Reply by matlabbe** [reply]
Changing any parameter under `Core Parameters` view won't trigger automatically an update of something. For example, if you change ICP parameters, they will be used when you do a "Refine" constraint (if `Reg/Strategy` is 1 or 2).

If you want to reprocess always everything after changing a parameter, the best is to use `rtabmap-reprocess` tool, which will then reprocess the whole database with your new parameters.

```
rtabmap-reprocess --help
```
